src/lib/context.ts

```diff
@@ -29,11 +31,24 @@ export interface Context {
     // may be considered equivalent.
     partOfName?: boolean;
     /**
-     * visitedNodes stores any nodes visited by rule 2F to
-     * ensure that any node is visited at most once. This
+     * visitedNodes stores any nodes visited recursively by rule 2F
+     * to ensure that any node is visited at most once. This
      * prevents infinite cycles during node traversal.
      */
     visitedNodes: Node[];
+
+    // TODO: Look into merging 'nodesUsed' and 'visitedNodes'
+
+    /**
+     * nodesUsed stores any Nodes used during the name computation
+     * algorithm, including the original target node.
+     */
+    nodesUsed: Set<Node>;
```

Comment:
`visitedNodes` and `nodesUsed` might appear to be the same thing here but replacing `visitedNodes` with `nodesUsed` resulted in ~1/3 of our unit tests failing. I'm going to look into merging these, or possibly just changing `visitedNodes`' name to further distinguish the two.